### PR TITLE
[2080] Add an inital label value when editing tree items label

### LINKED ISSUE #2080: Add an inital label value when editing tree items label
- EditingContextRepresentationDataFetcher will filter on IRepresentation unstead of ISemanticRepresentation to be able to support TreeDescription. As such targetObjectId is removed from RepresentationMetadata.

- The GraphQl input TreeEventInput now have an attribute treeId that start with "explorer://" when it's the explorer.

- A React component TreeItemDirectEditInput and the interface IInitialDirectEditTreeItemLabelProvider on the back-end are created.

### PR DESCRIPTION
**Pull request by mcharfadi**
Bug: https://github.com/eclipse-sirius/sirius-components/issues/2080

- EditingContextRepresentationDataFetcher will filter on IRepresentation unstead of ISemanticRepresentation to be able to support TreeDescription. As such targetObjectId is removed from RepresentationMetadata.
- The GraphQl input TreeEventInput now have an attribute treeId that start with "explorer://" when it's the explorer.
- A React component TreeItemDirectEditInput and the interface IInitialDirectEditTreeItemLabelProvider on the back-end are created.

### CODE REVIEW

File: integration-tests/cypress/e2e/project/edit/explorer.cy.js [hunk 1]
```diff
@@ -67,36 +67,41 @@ describe('/projects/:projectId/edit - Explorer', () => {
     cy.getByTestId('selected').contains('robot');
   });
 
+  const option = {
+    delay: 200,
+  };
+
```
Comment (sbegaudeau) [inline integration-tests/cypress/e2e/project/edit/explorer.cy.js]: Don't use a delay, improve instead the test. This will not work all the time because sometime, what you are waiting for will arrive in 300ms.

File: integration-tests/cypress/e2e/project/edit/document-context-menu.cy.js [hunk 2]
```diff
@@ -79,8 +79,8 @@ describe('/projects/:projectId/edit - Document Context Menu', () => {
     cy.getByTestId('robot-more').click();
     cy.getByTestId('treeitem-contextmenu').findByTestId('rename-tree-item').click();
 
-    cy.getByTestId('name-edit').type('renamed-robot{enter}');
-    cy.getByTestId('renamed-robot').should('exist');
+    cy.getByTestId('name-edit').type('.renamed{enter}');
+    cy.getByTestId('robot.renamed').should('exist');
```
Comment (sbegaudeau) [inline integration-tests/cypress/e2e/project/edit/document-context-menu.cy.js]: Why does the test needs to be changed? Did the behavior change? It should not but from what I understand here, it seems that when you start typing in the textfield, you are not replacing the original content but instead appending to the original content, is that the case?

File: CHANGELOG.adoc [hunk 3]
```diff
@@ -663,6 +663,7 @@ When closed, clicking on any of the views' icon will re-open the panel to make t
 - https://github.com/eclipse-sirius/sirius-components/issues/1201[#1201] [charts] Prepare support for charts in Sirius Components
 - https://github.com/eclipse-sirius/sirius-components/issues/1180[#1180] [diagram] Add support for the dynamic computation of the connector tools to control the tools displayed in the contextual menu.
 - https://github.com/eclipse-sirius/sirius-components/issues/1212[#1212] [form] Add support for styling of the View DSL widgets
+- https://github.com/eclipse-sirius/sirius-components/issues/2080[#2080] [tree] Add direct edit support on trees items.
```
Comment (sbegaudeau) [inline CHANGELOG.adoc]: Direct edit support on tree items (not trees items) has been there for a long time, this is not what this commit brings. Could you please update the title of the commit and the description of the change here to match what your contribution is improving?

Comment (sbegaudeau) [reply CHANGELOG.adoc]: Don't forget to describe the API breaks that are coming with this contributions in the dedicated section (such as RepresentationMetadata)

File: packages/core/backend/sirius-components-collaborative/src/main/java/org/eclipse/sirius/components/collaborative/editingcontext/EditingContextEventProcessorRegistry.java [hunk 4]
```diff
@@ -116,7 +113,6 @@ public void disposeEditingContextEventProcessor(String editingContextId) {
         this.logger.trace("Editing context event processors count: {}", this.editingContextEventProcessors.size());
     }
 
-    @PreDestroy
```
Comment (sbegaudeau) [inline packages/core/backend/sirius-components-collaborative/src/main/java/org/eclipse/sirius/components/collaborative/editingcontext/EditingContextEventProcessorRegistry.java]: Why is that class modified? I don't see any link with the topic of the commit

File: packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/editingcontext/EditingContextRepresentationDataFetcher.java [hunk 5]
```diff
@@ -66,22 +64,20 @@ public DataFetcherResult<RepresentationMetadata> get(DataFetchingEnvironment env
...
         var representationMetadata = this.editingContextEventProcessorRegistry.getEditingContextEventProcessors().stream()
                     .flatMap(editingContextEventProcessor -> editingContextEventProcessor.getRepresentationEventProcessors().stream())
                     .filter(editingContextEventProcessor -> editingContextEventProcessor.getRepresentation().getId().equals(representationId))
                     .map(IRepresentationEventProcessor::getRepresentation)
-                    .filter(ISemanticRepresentation.class::isInstance)
-                    .map(ISemanticRepresentation.class::cast)
-                    .map((ISemanticRepresentation representation) -> {
+                    .filter(IRepresentation.class::isInstance)
+                    .map(IRepresentation.class::cast)
+                    .map((IRepresentation representation) -> {
```
Comment (sbegaudeau) [inline packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/editingcontext/EditingContextRepresentationDataFetcher.java]: `IRepresentationEventProcessor#getRepresentation` already returns an `IRepresentation`, you don't need to filter it, cast it nor type it after that.

File: packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/editingcontext/EditingContextRepresentationDataFetcher.java [hunk 6]
```diff
@@ -101,14 +97,7 @@ public DataFetcherResult<RepresentationMetadata> get(DataFetchingEnvironment env
...
     private RepresentationMetadata toRepresentationMetadata(IRepresentation representation) {
-        // @formatter:off
-        String targetObjectId = Optional.of(representation)
-                .filter(ISemanticRepresentation.class::isInstance)
-                .map(ISemanticRepresentation.class::cast)
-                .map(ISemanticRepresentation::getTargetObjectId)
-                .orElse(null);
-        // @formatter:on
-        return new RepresentationMetadata(representation.getId(), representation.getKind(), representation.getLabel(), representation.getDescriptionId(), targetObjectId);
+        return new RepresentationMetadata(representation.getId(), representation.getKind(), representation.getLabel(), representation.getDescriptionId());
```
Comment (sbegaudeau) [inline packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/editingcontext/EditingContextRepresentationDataFetcher.java]: You could get rid of this method since it has no real purpose anymore and inline the code.

File: packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/editingcontext/EditingContextRepresentationsDataFetcher.java [hunk 7]
```diff
@@ -90,14 +88,7 @@ public Connection<RepresentationMetadata> get(DataFetchingEnvironment environmen
...
     private RepresentationMetadata toRepresentationMetadata(IRepresentation representation) {
-        // @formatter:off
-        String targetObjectId = Optional.of(representation)
-                .filter(ISemanticRepresentation.class::isInstance)
-                .map(ISemanticRepresentation.class::cast)
-                .map(ISemanticRepresentation::getTargetObjectId)
-                .orElse(null);
-        // @formatter:on
-        return new RepresentationMetadata(representation.getId(), representation.getKind(), representation.getLabel(), representation.getDescriptionId(), targetObjectId);
+        return new RepresentationMetadata(representation.getId(), representation.getKind(), representation.getLabel(), representation.getDescriptionId());
```
Comment (sbegaudeau) [inline packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/editingcontext/EditingContextRepresentationsDataFetcher.java:91]: Same comment, the method is not useful anymore

File: packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/representation/RepresentationMetadataDescriptionDataFetcher.java [hunk 8]
```diff
@@ -42,6 +48,8 @@
 @QueryDataFetcher(type = "RepresentationMetadata", field = "description")
 public class RepresentationMetadataDescriptionDataFetcher implements IDataFetcherWithFieldCoordinates<CompletableFuture<IRepresentationDescription>> {
 
+    static final BiFunction<VariableManager, String, IStatus> FAKE_RENAME_HANDLER = (a, b) -> new Failure("");
```
Comment (sbegaudeau) [inline packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/representation/RepresentationMetadataDescriptionDataFetcher.java]: Alway specify the visibility of the fields, on top of that, why aren't you inlining this directly? Why a static variable?

File: packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/representation/RepresentationMetadataDescriptionDataFetcher.java [hunk 9]
```diff
@@ -64,23 +92,27 @@ public CompletableFuture<IRepresentationDescription> get(DataFetchingEnvironment
...
+        if (Objects.equals(PropertiesEventProcessorFactory.DETAILS_VIEW_ID, representationMetadata.getDescriptionId()) || (Objects.equals(TreeEventProcessorFactory.DETAILS_VIEW_ID, representationMetadata.getDescriptionId()))) {
             /*
              * The FormDescription used for the details view can not be found by
              * IRepresentationDescriptionSearchService, but we can get away by returning a fake one with the same id as
              * no GraphQL query actually needs to see its content. We need to return *something* with the correct id
              * only to allow GraphQL resolution to continue on queries like "completionProposals" defined on
              * FormDescription.
              */
-            result = Mono.just(FAKE_DETAILS_DESCRIPTION).toFuture();
+            if (Objects.equals(PropertiesEventProcessorFactory.DETAILS_VIEW_ID, representationMetadata.getDescriptionId())) {
```
Comment (sbegaudeau) [inline packages/sirius-web/backend/sirius-web-graphql/src/main/java/org/eclipse/sirius/web/graphql/datafetchers/representation/RepresentationMetadataDescriptionDataFetcher.java:98]: Could you update the comment above since it does not strictly apply to the details view anymore

File: packages/sirius-web/backend/sirius-web-services/src/main/java/org/eclipse/sirius/web/services/explorer/ExplorerInitialDirectEditTreeItemLabelProvider.java [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+import org.eclipse.sirius.components.collaborative.trees.dto.InitialDirectEditElementLabelInput;
+import org.eclipse.sirius.components.collaborative.trees.dto.InitialDirectEditElementLabelSuccessPayload;
+import org.eclipse.sirius.components.core.api.IEditingContext;
+import org.eclipse.sirius.components.core.api.IPayload;
+import org.eclipse.sirius.components.trees.Tree;
+import org.eclipse.sirius.components.trees.TreeItem;
+import org.springframework.stereotype.Service;
+
+/**
+ * ExplorerInitialDirectEditTreeItemLabelProvider.
```
Comment (sbegaudeau) [inline packages/sirius-web/backend/sirius-web-services/src/main/java/org/eclipse/sirius/web/services/explorer/ExplorerInitialDirectEditTreeItemLabelProvider.java]: Could you provide a Javadoc with something a tiny bit more useful than just the name of the class (especially since that may change over time). Something like `Used to compute the initial label to display when the direct edit of a tree item of the explorer is triggered on the frontend`

File: packages/sirius-web/backend/sirius-web-services/src/main/java/org/eclipse/sirius/web/services/representations/RepresentationService.java [hunk 11]
```diff
@@ -50,6 +51,12 @@
 @Service
 public class RepresentationService implements IRepresentationService, IRepresentationPersistenceService, IDanglingRepresentationDeletionService {
 
+    public static final String EXPLORER_DESCRIPTION_ID = UUID.nameUUIDFromBytes("explorer_tree_description".getBytes()).toString();
+
+    public static final String EXPLORER_DOCUMENT_KIND = "siriusComponents://representation?type=Tree";
+
+    public static final String EXPLORER_NAME = "Explorer";
```
Comment (sbegaudeau) [inline packages/sirius-web/backend/sirius-web-services/src/main/java/org/eclipse/sirius/web/services/representations/RepresentationService.java]: It would be better to have those contants in the class providing the explorer instead of copying them here since they will change over time.

File: packages/trees/backend/sirius-components-collaborative-trees/src/main/java/org/eclipse/sirius/components/collaborative/trees/TreeEventProcessorFactory.java [hunk 12]
```diff
@@ -41,6 +42,8 @@
 @Service
 public class TreeEventProcessorFactory implements IRepresentationEventProcessorFactory {
 
+    public static final String DETAILS_VIEW_ID = UUID.nameUUIDFromBytes("explorer_tree_description".getBytes()).toString();
```
Comment (sbegaudeau) [inline packages/trees/backend/sirius-components-collaborative-trees/src/main/java/org/eclipse/sirius/components/collaborative/trees/TreeEventProcessorFactory.java]: - This is not related to the details view, be more careful while copying code
- This is not related to the explorer, the explorer is provided by a class in Sirius Web named `ExplorerDescriptionProvider`
- You are copying the same thing in very different locations, just reference what's in `ExplorerDescriptionProvider`

File: packages/trees/backend/sirius-components-collaborative-trees/src/main/java/org/eclipse/sirius/components/collaborative/trees/api/TreeConfiguration.java [hunk 13]
```diff
@@ -31,7 +31,7 @@ public class TreeConfiguration implements IRepresentationConfiguration {
 
     public TreeConfiguration(String editingContextId, List<String> expanded) {
         String uniqueId = editingContextId + expanded.toString();
-        this.treeId = UUID.nameUUIDFromBytes(uniqueId.getBytes()).toString();
+        this.treeId = "explorer://" + UUID.nameUUIDFromBytes(uniqueId.getBytes()).toString();
```
Comment (sbegaudeau) [inline packages/trees/backend/sirius-components-collaborative-trees/src/main/java/org/eclipse/sirius/components/collaborative/trees/api/TreeConfiguration.java:34]: I can understand why this is the shortest way to do what you need but this will not work forever., we will have to explore a better way to specify the identifier of the tree... It can be done later since there are no need to create additional work for this commit.

File: packages/trees/frontend/sirius-components-trees/src/treeitems/TreeItem.tsx [hunk 14]
```diff
@@ -181,6 +142,7 @@ export const TreeItem = ({
           showContextMenu: true,
           menuAnchor: currentTarget,
           editingMode: false,
+          editingkey: '',
```
Comment (sbegaudeau) [inline packages/trees/frontend/sirius-components-trees/src/treeitems/TreeItem.tsx]: Could you instead use prevState to stop calculating everything from scratch here and in the next two examples of this PR

File: packages/trees/frontend/sirius-components-trees/src/treeitems/TreeItemDirectEditInput.tsx [hunk 15]
```diff
@@ -0,0 +1,164 @@
...
+        representationId: treeId,
+        treeItemId: treeItemId,
+      },
+    }
+  );
+
+  useEffect(() => {
+    if (
+      !initialLabelTreeItemLoading &&
+      !initialLabelTreeItemItemError &&
```
Comment (sbegaudeau) [inline packages/trees/frontend/sirius-components-trees/src/treeitems/TreeItemDirectEditInput.tsx]: - Only test if you have some data, no need for the loading value here
- Display an error if there is an error (and don't forget that you can have both an error and some data)

File: packages/trees/frontend/sirius-components-trees/src/treeitems/TreeItemDirectEditInput.tsx [hunk 16]
```diff
@@ -0,0 +1,164 @@
...
+  treeId,
+  treeItemId,
+  refParent,
+  editingkey,
+  onClose,
+}: TreeItemDirectEditInputProps) => {
+  const [state, setState] = useState({
+    newLabel: '',
+    editingkey: editingkey,
+  });
```
Comment (sbegaudeau) [inline packages/trees/frontend/sirius-components-trees/src/treeitems/TreeItemDirectEditInput.tsx]: Type your state (you should have something like `useState<TreeItemDirectEditInputState>({...});`

File: packages/trees/frontend/sirius-components-trees/src/treeitems/TreeItemDirectEditInput.tsx [hunk 17]
```diff
@@ -0,0 +1,164 @@
...
+
+  const [renameTreeItem, { loading: renameTreeItemLoading, data: renameTreeItemData, error: renameTreeItemError }] =
+    useMutation(renameTreeItemMutation);
+
+  useEffect(() => {
+    if (!renameTreeItemLoading && !renameTreeItemError && renameTreeItemData?.renameTreeItem) {
+      const { renameTreeItem } = renameTreeItemData;
+      if (renameTreeItem.__typename === 'SuccessPayload') {
+        onClose();
+      }
```
Comment (sbegaudeau) [inline packages/trees/frontend/sirius-components-trees/src/treeitems/TreeItemDirectEditInput.tsx]: - The loading is not useful, you may not be loaded and may not have errors and still not have any data so test using data only
- Consider errors properly and display them like in any other React components

File: packages/trees/frontend/sirius-components-trees/src/treeitems/TreeItemDirectEditInput.tsx [hunk 18]
```diff
@@ -0,0 +1,164 @@
...
+    }
+  };
+
+  const onFocusIn = (event) => event.target.select();
+
+  const onFocusOut = (event) => {
+    //Prevents calling doRename when pressing Escape
+    if (!event.relatedTarget || event.relatedTarget != refParent) {
+      doRename();
+    }
```
Comment (sbegaudeau) [inline packages/trees/frontend/sirius-components-trees/src/treeitems/TreeItemDirectEditInput.tsx]: - Why not detect `Esc`?
- Do you really need the reference of the parent here?

File: packages/trees/frontend/sirius-components-trees/src/treeitems/TreeItemDirectEditInput.types.ts [hunk 19]
```diff
@@ -0,0 +1,47 @@
...
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+export interface TreeItemDirectEditInputProps {
+  editingContextId: string;
+  treeId: string;
+  treeItemId: string;
+  editingkey: string;
+  refParent: any;
```
Comment (sbegaudeau) [inline packages/trees/frontend/sirius-components-trees/src/treeitems/TreeItemDirectEditInput.types.ts]: Why any, if you really need that, you can type it

File: packages/trees/frontend/sirius-components-trees/src/views/ExplorerView.tsx [hunk 20]
```diff
@@ -170,6 +170,7 @@ export const ExplorerView = ({ editingContextId, selection, setSelection, readOn
       variables: {
         input: {
           id,
+          treeId: 'explorer',
```
Comment (sbegaudeau) [inline packages/trees/frontend/sirius-components-trees/src/views/ExplorerView.tsx:173]: Why not `tree.id`?

**Review by sbegaudeau — CHANGES_REQUESTED** [review]

File: integration-tests/cypress/e2e/project/edit/document-context-menu.cy.js (hunk 2 above)
Comment (mcharfadi) [reply integration-tests/cypress/e2e/project/edit/document-context-menu.cy.js]: Yes, ExplorerInitialDirectEditTreeItemLabelProvider in sirius web package return the current label, should I return an empty string ?

File: packages/trees/backend/sirius-components-collaborative-trees/src/main/java/org/eclipse/sirius/components/collaborative/trees/TreeEventProcessorFactory.java (hunk 12 above)
Comment (mcharfadi) [reply packages/trees/backend/sirius-components-collaborative-trees/src/main/java/org/eclipse/sirius/components/collaborative/trees/TreeEventProcessorFactory.java]: The explorer in ExplorerDescriptionProvider is using this ID. I can't do it the other way around.

File: packages/trees/frontend/sirius-components-trees/src/treeitems/TreeItemDirectEditInput.tsx (hunk 18 above)
Comment (mcharfadi) [reply packages/trees/frontend/sirius-components-trees/src/treeitems/TreeItemDirectEditInput.tsx]: This event is not fired when the user press Esc but when the component unmount after onClose() and onBlur is called.

File: packages/trees/frontend/sirius-components-trees/src/views/ExplorerView.tsx (hunk 20 above)
Comment (mcharfadi) [reply packages/trees/frontend/sirius-components-trees/src/views/ExplorerView.tsx:173]: What do you mean ?
This is for TreeEventInput, so we know it represents the explorer
`
input TreeEventInput {
  id: ID!
  treeId: String!
  editingContextId: ID!
  expanded: [String!]!
}
`

File: integration-tests/cypress/e2e/project/edit/document-context-menu.cy.js (hunk 2 above)
Comment (mcharfadi) [reply integration-tests/cypress/e2e/project/edit/document-context-menu.cy.js]: ExplorerInitialDirectEditTreeItemLabelProvider now return an empty initialLabel.

File: packages/trees/frontend/sirius-components-trees/src/views/ExplorerView.tsx (hunk 20 above)
Comment (sbegaudeau) [reply packages/trees/frontend/sirius-components-trees/src/views/ExplorerView.tsx:173]: Indeed it would not make sense to rely on tree.id since we wouldn't have it at that time but we still have an issue with the lifecycle of the identity of the explorer but that's not an issue for now so we don't need to try to solve that for now. `"explorer"` will be fine for now.

File: packages/sirius-web/backend/sirius-web-services/src/main/java/org/eclipse/sirius/web/services/explorer/ExplorerInitialDirectEditTreeItemLabelProvider.java [hunk 21]
```diff
@@ -0,0 +1,49 @@
...
+    public static final String EXPLORER_NAME = "Explorer";
+
+    @Override
+    public boolean canHandle(Tree tree) {
+        return tree.getId().startsWith("explorer://");
+    }
+
+    @Override
+    public IPayload handle(IEditingContext editingContext, Tree tree, InitialDirectEditElementLabelInput input) {
+        return new InitialDirectEditElementLabelSuccessPayload(input.id(), "");
```
Comment (sbegaudeau) [inline packages/sirius-web/backend/sirius-web-services/src/main/java/org/eclipse/sirius/web/services/explorer/ExplorerInitialDirectEditTreeItemLabelProvider.java]: You need to return here the current label of the selected tree item

File: packages/trees/backend/sirius-components-collaborative-trees/src/main/java/org/eclipse/sirius/components/collaborative/trees/handlers/InitialDirectEditTreeItemLabelEventHandler.java [hunk 22]
```diff
@@ -0,0 +1,79 @@
...
+        return treeInput instanceof InitialDirectEditElementLabelInput;
+    }
+
+    @Override
+    public void handle(One<IPayload> payloadSink, Many<ChangeDescription> changeDescriptionSink, IEditingContext editingContext, Tree tree, ITreeInput treeInput) {
+        IPayload payload = new InitialDirectEditElementLabelSuccessPayload(treeInput.id(), "");
+        ChangeDescription changeDescription = new ChangeDescription(ChangeKind.NOTHING, treeInput.representationId(), treeInput);
+
+        if (treeInput instanceof InitialDirectEditElementLabelInput input) {
+            Optional<IInitialDirectEditTreeItemLabelProvider> optionalPathProvider = this.initialDirectEditTreeItemLabelProvider.stream().filter(treePathProvider -> treePathProvider.canHandle(tree)).findFirst();
```
Comment (pcdavid) [inline packages/trees/backend/sirius-components-collaborative-trees/src/main/java/org/eclipse/sirius/components/collaborative/trees/handlers/InitialDirectEditTreeItemLabelEventHandler.java:65]: This still uses the names `treePathProvider` and `optionalPathProvider` (from `TreePathEventHandler` which probaly served as the starting point).

**Review by pcdavid — CHANGES_REQUESTED** [review]

File: packages/sirius-web/backend/sirius-web-services/src/main/java/org/eclipse/sirius/web/services/explorer/ExplorerInitialDirectEditTreeItemLabelProvider.java [hunk 23]
```diff
@@ -0,0 +1,73 @@
...
+    public static final String EXPLORER_NAME = "Explorer";
+
+    @Override
+    public boolean canHandle(Tree tree) {
+        return tree.getId().startsWith("explorer://");
+    }
+
+    @Override
+    public IPayload handle(IEditingContext editingContext, Tree tree, InitialDirectEditElementLabelInput input) {
+        String initialLabel = tree.getChildren().stream()
```
Comment (pcdavid) [inline packages/sirius-web/backend/sirius-web-services/src/main/java/org/eclipse/sirius/web/services/explorer/ExplorerInitialDirectEditTreeItemLabelProvider.java:48]: I have not tested, but it's possible this might be simplifiable into something like:

```java
new TreeQueryService().findTreeItem(tree, input.treeItemId()).map(TreeItem::getLabel).orElse("")
```

Comment (mcharfadi) [reply packages/sirius-web/backend/sirius-web-services/src/main/java/org/eclipse/sirius/web/services/explorer/ExplorerInitialDirectEditTreeItemLabelProvider.java:48]: Yes it worked, thanks you

**Review by pcdavid — APPROVED** [review]

**Review by sbegaudeau — APPROVED** [review]
